config/management/src/secure_backend.rs

```diff
@@ -0,0 +1,129 @@
+/// SafetyRulesBackend type to parse.
+struct SecureBackend {
+    backend: String,
+    parameters: HashMap<String, String>,
+}
+
+impl SecureBackend {
+    const BACKEND: &'static str = "backend";
+}
+
+impl From<&str> for SecureBackend {
+    fn from(s: &str) -> Self {
+        let mut parameters = s
+            .split(';')
+            .map(|pair| {
+                let kv = pair.split('=').collect::<Vec<_>>();
+                if kv.len() != 2 {
+                    panic!("Invalid kv found in backend: {:?}", kv);
+                }
+                (kv[0].into(), kv[1].into())
+            })
+            .collect::<HashMap<String, String>>();
+        let backend = parameters.remove(Self::BACKEND).expect("Missing backend");
+        Self {
+            backend,
+            parameters,
+        }
+    }
+}
+
+impl Into<SafetyRulesBackend> for SecureBackend {
+    fn into(mut self) -> SafetyRulesBackend {
+        let backend = match self.backend.as_ref() {
+            "disk" => {
+                let mut config = OnDiskStorageConfig::default();
+                config.set_data_dir(PathBuf::from(""));
+                let path = self.parameters.remove("path");
+                let path = path.expect("Unable to retrieve path from disk backend");
+                config.path = PathBuf::from(path);
+                SafetyRulesBackend::OnDiskStorage(config)
+            }
+            "memory" => SafetyRulesBackend::InMemoryStorage,
+            "vault" => {
+                let server = self.parameters.remove("server");
+                let server = server.expect("Missing server");
+                let token = self.parameters.remove("token");
+                let token = token.expect("Missing token");
+                SafetyRulesBackend::Vault(VaultConfig {
+                    default: false,
+                    namespace: self.parameters.remove("namespace"),
+                    server,
+                    // TODO(davidiw) Make this a path to a file
+                    token,
+                })
+            }
+            _ => panic!("Invalid backend: {}", self.backend),
+        };
+
+        if !self.parameters.is_empty() {
+            panic!("Found extra parameters for backend: {:?}", self.parameters);
+        }
+
+        backend
+    }
+}
+
+#[allow(dead_code)]
+pub fn storage(s: &str) -> Box<dyn Storage> {
+    let management_backend: SecureBackend = s.into();
+    let sr_backend: SafetyRulesBackend = management_backend.into();
+    (&sr_backend).into()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_memory() {
+        let memory = "backend=memory";
+        storage(memory);
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_memory_fail() {
+        let memory = "backend=memory;extra=stuff";
+        storage(memory);
+    }
+
+    #[test]
+    fn test_disk() {
+        let disk = "backend=disk;path=some_path";
```

Comment:
Nit: does the implementation currently support whitespace? i.e., what if I wanted to do:

"backend=vault ; server=http://127.0.0.1:8080 ; token=123456 ; namespace=test"

Whitespace just seems to make these long strings a lot more readable.

Reply:
no, it does not, worth thinking about the tradeoffs... personally not a big fan of whitespace :P

Reply:
Okay.. personally I find it somewhat hard to parse, but it's only really in the vault case.